### DISCUSSION TITLE
PyAEDT RMxprt to create "user define slot" under standard motor

### DISCUSSION THREAD

**Discussion by ChunLai-CHEN** [ansys/pyaedt#5136] [Q&A] [answered]
hi there, 

I want to create "user define slot" of ASSM under standard motor of RMxprt by pyAEDT. And it does not work. 

Could you help to show how to use this "user define slot" option by pyAEDT please.

![image](https://github.com/user-attachments/assets/113e4bb6-7c47-469a-8386-ed8c9d4d60c8)

![image](https://github.com/user-attachments/assets/55f84d82-4950-4b9e-9a5f-df045630463b)

**Comment by gmalinve** [comment]
Hi @ChunLai-CHEN, may I ask which AEDT and PyAEDT versions are you using?

**Reply by ChunLai-CHEN** [reply]
2024.2@AEDT, and 0.9.10@PyAEDT, FYI

**Comment by gmalinve** [comment]
Hi @ChunLai-CHEN , could you please upgrade to latest version of PyAEDT and try again please? thanks!

Also, what's "Segment1" for you? could you provide your script so I can have a better overview of your issue?

**Reply by ChunLai-CHEN** [reply]
hi @gmalinve, first of all, many thanks for your help. 

Regrading the issue, that I would like learn to use PyAEDT to create a RMxprt model based on PyAEDT example, "create and export motor", see the image below. 
![image](https://github.com/user-attachments/assets/8eadddc6-6817-4e9c-b9ae-110ae67cf227)

For the stator definition, the example gives the slot type number and other children properties, see the image below. However, I would like to use other slot shape need to use "user define slot" rather than slot type number.
![image](https://github.com/user-attachments/assets/504f9bbf-fe8b-410e-98cb-aa154e5b7472)

Based on the PyAEDT example, I change the slot type number to "UserDefinedSlot" and software could recognize the user define type without error. The next step I need to define the segments of the slot, which is "Segment" definition comes from. 
![image](https://github.com/user-attachments/assets/6d657e29-1cf1-4814-8984-10ff3b1fe6de)
![image](https://github.com/user-attachments/assets/2ccff505-2e64-43f4-84e8-8ca81db03e1b)
![image](https://github.com/user-attachments/assets/ef42e2a1-8b77-4d22-9fdf-81dd1b518f72)

I could not find reference of **child properties** of this "UserDefinedSlot", such as user define slot type and segment definition using PyAEDT. Thanks for your help in advance.
![image](https://github.com/user-attachments/assets/cc4b51dc-cc27-4b19-ae4e-9143f0fd6f6d)
![image](https://github.com/user-attachments/assets/84f38192-f925-428d-8f2f-41e796da538d)

**Comment by gmalinve** [comment]
Hi @ChunLai-CHEN,

Thanks for your detailed explanation.
If you want to set a user defined slot through pyaedt you should do it through the PyAEDT stator obejct as: 
`app.stator["Slot Type"] = "UserDefinedSlot"`
Then you will be able to see in AEDT:
![image](https://github.com/user-attachments/assets/107ddc6f-3da7-4591-abc7-e02ed170f278)
Now, if you evaluate the stator properties you can see that _children_ now has the _UserDefSymmetricSlot_. Look at my evaluator (I use PyCharm as my IDE): 
![image](https://github.com/user-attachments/assets/5d4cfd58-83e8-40a1-a07d-37d1cd1be57f)
Through children you can then access to your segments as:
`app.stator.properties.children["UserDefSymmetricSlot"].children["Segment1"]` 
and consequentially its properties:
![image](https://github.com/user-attachments/assets/1444cc29-5f61-43c3-ae0c-2964f0dc8a02)

Hope this helps,
Kind regards,

Giulia

**Comment by ChunLai-CHEN** [comment]
hi @gmalinve, 

Thanks for your share, really helpful. 

 Try your method to find the expression of the "user defined slot", and get some improvement that "segment1" is fine. However it does not work for other segments. Python gives the error message of "Segment2"

![image](https://github.com/user-attachments/assets/55a9951c-65c8-4e5a-83ac-9a1dd1dd3797)
![image](https://github.com/user-attachments/assets/05c4809d-ecbf-486e-b322-ed786f98bcb7)
![image](https://github.com/user-attachments/assets/5f4708b3-4954-49a4-8029-92e1d0089b91)

In AEDT GUI mode, there is option "Append Segment" to add next segment manually, however I do not familiar with the expression of children properties of "user define slot", and seems my code about add other segments does not work. 

![image](https://github.com/user-attachments/assets/657d0ebc-044c-4d8e-b9e6-01aa8777a43d)

It would be grateful you share some code example how to add other segments to make the user define slot complete. 

Best,

CC

**Comment by gmalinve** [comment]
Hi @ChunLai-CHEN, you get this error because you are trying to access a segment that doesn't exist yet in your design (if I look at your first screenshot of AEDT GUI). 
We don't have a method yet to create segments.
If you try to create segments manually then you can access their properites with children as I showed you in my previous comment.

**Comment by ChunLai-CHEN** [comment]
Hi @gmalinve，will PyAEDT could create segments in future version?

**Comment by gmalinve** [comment, marked as answer]
Hi @ChunLai-CHEN,

PyAEDT is an open repository which means that everyone can contribute through issues and pull requests.
There's no team behind it but rather a community.
If this is a high priority matter please open an issue here: https://github.com/ansys/pyaedt/issues and someone will take care of it eventually.
Or if you feel comfortable you can implement it yourself and someone from the community will review it.

Thanks,
Regards,

Giulia
